smoke_test/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+# First copy the CMakeList.txt so we can build the test
+file(COPY ${PACKAGE_SOURCE_DIR}/smoke_test/CMakeLists.txt.in DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/source)
```

Comment:
Consider renaming: smoke_test -> spack_test as clearer meaning.

Reply:
I want to be able to perform this test even with a CMake build so I think that spack_test would actually become more confusing in that case... I'm not against renaming it something clearer though, maybe installation_test?